src/main/java/org/broadinstitute/hellbender/tools/spark/sv/SVKmerBase.java

```diff
@@ -0,0 +1,66 @@
+package org.broadinstitute.hellbender.tools.spark.sv;
+
+import com.esotericsoftware.kryo.Kryo;
+import com.esotericsoftware.kryo.io.Input;
+import com.esotericsoftware.kryo.io.Output;
+import org.broadinstitute.hellbender.utils.BaseUtils;
+import org.broadinstitute.hellbender.utils.Utils;
+
+//Superclass for SVKmer and SVKmerSmall
+
+public abstract class SVKmerBase {
+
+    public enum Base {
+        A(0L),
+        C(1L),
+        G(2L),
+        T(3L);
+
+        public final long value;
+
+        Base(final long value) { this.value = value; }
+    }
+
+    // Lookup table for reverse-complementing each possible byte value.
+    // Each pair of bits represents a base, so you have to reverse bits pairwise and then invert all bits.
+    // This is most quickly and easily done with a lookup table.
+    protected static final long[] BYTEWISE_REVERSE_COMPLEMENT;
+    static {
+        BYTEWISE_REVERSE_COMPLEMENT = new long[256];
+        for ( int idx = 0; idx != 256; ++idx ) {
+            BYTEWISE_REVERSE_COMPLEMENT[idx] = reverseComplementByteValueAsLong(idx);
+        }
+    }
+
+    protected abstract void serialize(final Kryo kryo, final Output output );
```

Comment:
Ought to remove this.  It's not really a feature of a kmer, but an implementation detail to be handled in each concrete class that needs it.